src/v0/destinations/braze/util.js

```diff
@@ -0,0 +1,181 @@
+
+  // Ref: https://www.braze.com/docs/user_guide/administrative/access_braze/braze_instances
+  if (destination.Config.dataCenter) {
+    const dataCenterArr = destination.Config.dataCenter.trim().split('-');
+    if (dataCenterArr[0].toLowerCase() === 'eu') {
+      endpoint = `https://rest.fra-${dataCenterArr[1]}.braze.eu`;
+    } else {
+      endpoint = `https://rest.iad-${dataCenterArr[1]}.braze.com`;
+    }
+  }
+  return endpoint;
+};
+
+const BrazeDedupUtility = {
+  async doLookup(inputs) {
+    const externalIds = [];
+    const aliasIds = [];
+    // eslint-disable-next-line no-restricted-syntax
+    for (const input of inputs) {
+      const { message } = input;
+      const brazeExternalId = getDestinationExternalID(message, 'brazeExternalId');
+      const userId = getFieldValueFromMessage(message, 'userIdOnly');
+      const anonymousId = get(message, 'anonymousId');
+      if (brazeExternalId) {
+        externalIds.push(brazeExternalId);
+      } else if (userId) {
+        externalIds.push(userId);
+      } else if (anonymousId) {
+        aliasIds.push(anonymousId);
+      }
+    }
+    const externalIdsToQuery = [...new Set(externalIds)];
+    const aliasIdsToQuery = [...new Set(aliasIds)];
+
+    const identifiers = [];
+    if (externalIdsToQuery.length > 0) {
+      externalIdsToQuery.forEach((externalId) => {
+        identifiers.push({
+          external_id: externalId,
+        });
+      });
+    }
+    if (aliasIdsToQuery.length > 0) {
+      aliasIdsToQuery.forEach((aliasId) => {
+        identifiers.push({
+          alias_name: aliasId,
+          alias_label: 'rudder_id',
+        });
+      });
+    }
+    const identfierChunks = _.chunk(identifiers, 50);
+
+    const chunkedUserData = await Promise.all(
+      identfierChunks.map(async (ids) => {
+        const externalIdentifiers = ids.filter((id) => id.external_id !== undefined);
+        const aliasIdentifiers = ids.filter((id) => id.alias_name !== undefined);
+
+        const startTime = Date.now();
+        const lookUpResponse = await httpPOST(
+          `${getEndpointFromConfig(inputs[0].destination)}/users/export/ids`,
+          {
+            external_ids: externalIdentifiers.map((extId) => extId.external_id),
+            user_aliases: aliasIdentifiers,
+          },
+          {
+            headers: {
+              Authorization: `Bearer ${inputs[0].destination.Config.restApiKey}`,
+            },
+            timeout: 10 * 1000,
+          },
+        );
+        const endTime = Date.now();
+        // TODO: Remove this log
+        console.log(
+          `Time taken to fetch user store: ${endTime - startTime} ms for ${ids.length} users`,
+        );
+        const processedLookUpResponse = processAxiosResponse(lookUpResponse);
+        const { users } = processedLookUpResponse.response;
+
+        return users;
+      }),
+    );
+
+    return _.flatMap(chunkedUserData);
+  },
+
+  enrichUserStore(users, store) {
```

Comment:
updateUserStore may be better name, also updateUserStore(store, users) is better signature

Reply:
addressed